docs/design/908-proposal-process.md

```diff
@@ -0,0 +1,410 @@
+
+   * Accept the proposal, or
+   * decline the proposal.
+
+[new-proposal]: https://github.com/grafana/alloy/issues/new?assignees=&labels=proposal&projects=&template=proposal.yaml
+
+### Scope
+
+The proposal process is intended for user-facing features, enhancements or any
+largely scoped internal change. Other changes typically do not need a proposal,
+such as minor code refactors. If a contributor is uncertain about whether a
+proposal is needed, it is preferable to create a proposal instead of jumping
+into a change that may be declined.
+
+Proposals may initially be brief, with just enough detail to explain what is
+being proposed, why it is being proposed, and any relevant details that may
+guide the public discussion towards consensus.
+
+Proposal authors may skip straight to creating a design document for their
+proposal if they know one will be needed. If in doubt, it's better to create a
+proposal issue first.
+
+Any existing issue can be turned into a proposal by adding a `proposal` label.
+
+### Compatibility
+
+Proposals that break [backward compatibility] are moved to a "Hold" state until
+work on the next major release begins. As major releases are infrequent, this
+avoids approved proposals from becoming obsolete before they are implemented.
+
+[backward compatibility]: https://grafana.com/docs/alloy/latest/introduction/backward-compatibility/
+
+### Design documents
+
+Proposals that require more careful consideration and explanation require a
+design document.
+
+* Design documents should be checked into the Alloy repository as
+  `docs/design/NNNN-name.md`, where `NNNN` is the GitHub issue number of the
+  proposal and `name` is a short descriptive name deliminated by hyphens.
+
+* Design documents should follow the template in `docs/design/template.md`.
+
+* Design documents should address any specific concerns raised during the
+  initial discussion.
+
+* Design documents should acknowledge and address alternative solutions, if
+  known, to the problem statement.
+
+Once a design document is created, the proposal author should:
+
+* Close the original proposal issue and link to the design document PR.
+* Update the proposal issue description with a link to the design document.
+
+Closing the original proposal issue in favor of the design document PR prevents
+fragmenting discussions as contributors work towards consensus on a proposal.
+
+Design documents do not track the state of the proposal; the proposal PR  is
+the source of truth.
+
+The PR for design documents are merged once the associated proposal becomes
+_Accepted_. If the proposal becomes _Declined_, the PR for the design document
+is closed.
+
+### Design document template
+
+```markdown
+# Proposal: [Title]
+
+* Author(s): [List of proposal authors]
+* Last updated: [Date]
+* Original issue: https://github.com/grafana/alloy/issues/NNNN
+
+## Abstract
+
+[A short summary of the proposal.]
+
+## Background
```

Comment:
I can change this, but I'm not sure I fully understand why the header name makes a significant difference to the template, since the description of the section describes that a problem statement should be included. Can you help me understand more before I change it?

Reply:
I think it's a nudge in the right direction and if someone skims through the template or gets honestly lost in the editorial process, then it's easier to notice that a section titled Problem doesn't describe the problem than it is to notice that a section titled Background doesn't describe the problem.